lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -2285,6 +2335,200 @@ prop_balanceTransactionUnresolvedInputs
         let dropped = map snd $ filter (not . fst) $ zip shouldKeep inputs
         pure (PartialTx tx inputs' redeemers, dropped)
 
+data BalanceTxGolden =
+    BalanceTxGoldenSuccess
+        { _walletCoin :: Coin
+        , _balancedFee :: Cardano.Lovelace
+        , _balancedMinFee :: Cardano.Lovelace
+        }
+    | BalanceTxGoldenFailure Coin String
```

Comment:
I guess the downside of having a record constructor here is that `_walletCoin` (and so on) are now partial functions that will fail at runtime if someone accidentally applies them to a `BalanceTxGoldenFailure`. On the other hand, this is a test, so it probably doesn't matter so much.

Reply:
> On the other hand, this is a test, so it probably doesn't matter so much.

Indeed. I thought it was worth it given that both `balancedFee` and `balancedMinFee` have the same type... However, maybe just using haddock comments isn't that bad:
```
BalanceTxGoldenSuccess
        Coin -- ^ Wallet balance
        Cardano.Lovelace -- ^ Fee
        Cardano.Lovelace -- ^ Minimum fee
```
particularly as I don't even use the record constructors.